File: LeetCode_1018/main.cpp
/**
    LeetCode_1018
    https://leetcode.com/problems/binary-prefix-divisible-by-5/
    MatrixDeity, 2022.
*/

#include <vector>

using namespace std;

class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& A) {
        int num = 0;
        vector<bool> res;
        res.reserve(A.size());
        for (auto currNum : A) {
            num = (2 * num + currNum) % 5;
            res.push_back(num == 0);
        }
        return res;
    }
};

File: LeetCode_1018/main.go
/**
  LeetCode_1018
  https://leetcode.com/problems/binary-prefix-divisible-by-5/
  MatrixDeity, 2025.
*/

package leetcode1018

func prefixesDivBy5(nums []int) []bool {
	res := make([]bool, 0, len(nums))
	x := 0

	for _, num := range nums {
		x = (x | num) % 5
		res = append(res, x == 0)
		x <<= 1
	}

	return res
}
